pandas/tests/series/indexing/test_indexing.py

```diff
@@ -768,3 +768,24 @@ def test_head_tail(test_data):
     assert_series_equal(test_data.series.head(0), test_data.series[0:0])
     assert_series_equal(test_data.series.tail(), test_data.series[-5:])
     assert_series_equal(test_data.series.tail(0), test_data.series[0:0])
+
+
+def test_reset_index_drop_errmsg():
+    # https://github.com/pandas-dev/pandas/issues/20925
+
+    # Check KeyError raised for series index where passed level name is missing
+    s = pd.Series(range(4))
+    with tm.assert_raises_regex(KeyError, 'must be same as name'):
+        s.reset_index('wrong', drop=True)
+    with tm.assert_raises_regex(KeyError, 'must be same as name'):
+        s.reset_index('wrong')
+
+    # Check KeyError raised for series where 'level' to be dropped is missing
+    s = pd.Series(range(4), index=pd.MultiIndex.from_product([[1, 2]] * 2))
+    with tm.assert_raises_regex(KeyError, 'not found'):
+        s.reset_index('wrong', drop=True)
+
+    # Check that .reset_index([],drop=True) doesn't fail
```

Comment:
This is fine but probably not in the right test: could you append it to ``test_reset_index_level`` (in ``pandas/tests/series/test_alter_axes.py``)?

Reply:
Done. Have moved it to function 'test_reset_index_level' outside the for loop